Pony/include/pony/Address.hpp

```diff
@@ -9,34 +9,33 @@
 #ifndef PONY_ADDRESS_HPP
 #define PONY_ADDRESS_HPP
 
-#include <pony/SharedMacros.hpp>
+#include <stdint.h>
 
 namespace pony
 {
 
-class PONY_EXPORT Address
+class Address
 {
-
 public:
     Address();
-    Address(unsigned int address);
-    Address(unsigned int address, unsigned short port);
-    Address(unsigned char fst, unsigned char snd, unsigned char thd, unsigned char fth, unsigned short port);
+    Address(uint32_t address);
+    Address(uint32_t address, uint16_t port);
+    Address(uint8_t fst, uint8_t snd, uint8_t thd, uint8_t fth, uint16_t port);
 
-    unsigned int GetAddress() const;
-    unsigned short GetPort() const;
+    uint32_t GetAddress() const;
+    uint16_t GetPort() const;
 
-    const bool operator == (const Address & rhs);
-    const bool operator != (const Address & rhs);
+    uint8_t GetAddressPart(uint32_t nr) const;
 
-protected:
+    bool operator == (const Address& rhs) const;
```

Comment:
Variables that can fit into the CPU registers are better to be passed by value. It won't break the operator overloading.

Reply:
oh no!
ok..